lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -477,6 +635,14 @@ void LowerClassesPass::runOnOperation() {
   hw::InnerSymbolNamespaceCollection namespaces;
   HierPathCache cache(circuit, symbolTable);
 
+  // Fill `shouldCreateClassMemo`.
+  for (auto moduleLike : circuit.getOps<FModuleLike>())
+    shouldCreateClassMemo.insert({moduleLike, false});
+  parallelForEach(
```

Comment:
Please also check the single-threaded performance.  Parallelization only improves walltime.  We need to improve absolute time also.

Reply:
I checked the single threaded performance on a large internal design, and this change leads to a >90% reduction in runtime in a single threaded context.